packages/amplify_auth_cognito/example/integration_test/user_attributes_test.dart

```diff
@@ -0,0 +1,235 @@
+const phoneNumberAttributeKey = 'phone_number';
+const givenNameAttributeKey = 'given_name';
+const emailVerifiedAttributeKey = 'email_verified';
+
+dynamic getAttributeValueFromList(
+  List<AuthUserAttribute> userAttributes,
+  String keyName,
+) {
+  return userAttributes
+      .firstWhere((attribute) => attribute.userAttributeKey == keyName)
+      .value;
+}
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  final username = generateUsername();
+  final password = generatePassword();
+  final email = generateEmail();
+  final phoneNumber = mockPhoneNumber;
+  final name = generateNameAttribute();
+
+  group('User Attributes', () {
+    setUpAll(() async {
+      await configureAuth();
+      await signOutUser();
+      await Amplify.Auth.signUp(
+          username: username,
+          password: password,
+          options: CognitoSignUpOptions(userAttributes: {
+            emailAttributeKey: email,
+            phoneNumberAttributeKey: phoneNumber,
+            nameAttributeKey: name
+          }));
+      await Amplify.Auth.signIn(username: username, password: password);
+    });
+
+    group('fetchUserAttributes', () {
+      testWidgets('should fetch a users attributes',
+          (WidgetTester tester) async {
+        var userAttributes = await Amplify.Auth.fetchUserAttributes();
+
+        var emailAttributeValue = getAttributeValueFromList(
+          userAttributes,
+          emailAttributeKey,
+        );
+        var phoneNumberAttributeValue = getAttributeValueFromList(
+          userAttributes,
+          phoneNumberAttributeKey,
+        );
+        var nameAttributeValue = getAttributeValueFromList(
+          userAttributes,
+          nameAttributeKey,
+        );
+
+        expect(emailAttributeValue, email);
+        expect(phoneNumberAttributeValue, phoneNumber);
+        expect(nameAttributeValue, name);
+      });
+    });
+
+    group('updateUserAttribute', () {
+      testWidgets('should update a single users attribute',
+          (WidgetTester tester) async {
+        final updatedName = generateNameAttribute();
+        var res = await Amplify.Auth.updateUserAttribute(
+          userAttributeKey: nameAttributeKey,
+          value: updatedName,
+        );
+
+        expect(res.nextStep.updateAttributeStep, 'DONE');
+
+        var userAttributes = await Amplify.Auth.fetchUserAttributes();
+
+        var nameAttributeValue = getAttributeValueFromList(
+          userAttributes,
+          nameAttributeKey,
+        );
+
+        expect(nameAttributeValue, updatedName);
+      });
+
+      testWidgets('should require confirmation for an email attribute',
+          (WidgetTester tester) async {
+        final updatedEmail = generateEmail();
+        var res = await Amplify.Auth.updateUserAttribute(
+          userAttributeKey: emailAttributeKey,
+          value: updatedEmail,
+        );
+
+        expect(res.nextStep.updateAttributeStep, 'CONFIRM_ATTRIBUTE_WITH_CODE');
```

Comment:
in ios and android this is failing bc is "DONE", dk if there is some config thing to do or something like that.

Reply:
I ran through this again just to be sure and I am seeing it pass. You can remove the requirement to confirm email addresses in Cognito (see screenshot). It should be set to email by default I believe. I don't think anything in the integration test provisioning would change that. When I set MFA to "optional" and update "Which attributes do you want to verify?" to "No verification" I also see this fail with "DONE".

Do you want to see what yours is set to? Maybe you had disabled it while setting up the integration tests initially? You can find the option under "MFA and verifications" in the left hand menu from the Cognito user pool for your project.

Screenshot from Cognito:

![Screen Shot 2021-07-13 at 3 21 18 PM](https://user-images.githubusercontent.com/20613561/125512713-86c20e8a-e4da-4eb6-9dc1-521672fab185.png)